### ISSUE TITLE
Implement a game loop

### ISSUE THREAD

**Issue by davidwengier** [jamesformica/Trains#7]

**Comment by ghost** [comment]
I'd love to see a processing/rendering time independent game loop, so speed is calculated by how quick we can run the loop, meaning the user experience (train speed) is the same no matter if it is rendering at 2fps or 30fps.

Also implement a redraw rate limiter, no point in drawing over 30fps.

**Comment by davidwengier** [comment]
I think a draw loop and a game loop being separate makes sense. It would help with debugging :)

**Comment by ghost** [comment]
True. I think one hit timers would be the best bet. You run RenderBoard() and calculate time till next render is required. That way you can limit frames, 33ms - TimeToRender = Timer interval, giving max 30fps, and limit minimum redraw, eg, if TimeToRender >= 33ms, still sleep for 10ms or so, lowering the FPS but still giving CPU time to other operations.

**Comment by ghost** [comment]
TL;DR: Yes, 2 timers.

**Comment by davidwengier** [comment]
https://www.youtube.com/watch?v=KDFclOrzXeQ
